### PR TITLE
TM-3088, sort ass history by start date

### PR DESCRIPTION
**Pull request by M00staff**
Dual Merge: 
- [FE PR](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2640)

[Ticket](https://metaphase.atlassian.net/browse/TM-3088)

### CODE REVIEW

**Comment by scott062** [comment]
We can pass an orderby field to the query and let the DB give us sorted results - we don't have to manually sort it ourselves

**Comment by M00staff** [comment]
> We can pass an orderby field to the query and let the DB give us sorted results - we don't have to manually sort it ourselves

hmm - that doesn't seem to be working. 

i believe I'm doing it right
image

 - and it seems to work in swagger
image
image

**Comment by scott062** [comment]
Right - mock may not be able to filter on every field. Trust if it works in swagger then we should be okay

**Comment by M00staff** [comment]
> Right - mock may not be able to filter on every field. Trust if it works in swagger then we should be okay

derp - i wanna say i should have known that

just tested in dev, and this is working! 🎉

**Comment by scott062** [comment]
Just moved this up a level so the view determines the ordering jic we want the service to behave differently later. Sorting by another order etc

**Review by scott062 — APPROVED** [review]

File: talentmap_api/fsbid/services/assignment_history.py [hunk 1]
```diff
@@ -99,6 +99,7 @@ def convert_assignments_query(query):
         "rp.pageRows": int(query.get('limit', 1000)),
         "rp.filter": filters,
         "rp.columns": "asgperdetseqnum",
+        "rp.orderBy": services.sorting_values(query.get("ordering", "-assignment_start_date")),
```
Comment (scott062) [inline talentmap_api/fsbid/services/assignment_history.py:102]: Could've just added this line and not worried about hard coding it into the views but I like the deterministic pattern at the view so I don't have to sift through fallbacks

**Pull request merged**